File: lc1_TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> diffs = new HashMap<>();
        int result[] = new int[0];
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (diffs.get(diff) != null) {
                if (diffs.get(diff) < i) {
                    result = new int[] {diffs.get(diff), i};
                } else {
                    result = new int[] {diffs.get(diff), i};
                }
            }
            diffs.put(nums[i], i);
        }
        return result;
    }
}

File: lc1_TwoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        diffs = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diffs.get(diff) is not None:
                return [diffs.get(diff), i] if diffs.get(diff) < i else [i, diffs.get(diff)]
            diffs[nums[i]] = i